src/classes/RD2_ScheduleService.cls

```diff
@@ -81,6 +81,54 @@ public without sharing class RD2_ScheduleService {
         return newSchedules;
     }
 
+/***
+* @description Calculates the date of the next donation
+* @param rd Recurring Donation record
+* @return Date
+*/
+    public Date getNextDonationDate(npe03__Recurring_Donation__c rd) {
+        if (rd.Status__c == RD2_Constants.STATUS_CLOSED) {
+            return null;
+        }
+
+        return getNextDonationDate(getNewSchedules(rd));
+    }
+
+/***
+* @description Calculates the date of the next donation
+* @param schedule Recurring Donation Schedule record
+* @return Date
+*/
+    public Date getNextDonationDate(List<RecurringDonationSchedule__c> rdSchedules) {
+
+        Date nextDate;
+        Date previousDate;
+
+        for (RecurringDonationSchedule__c schedule : rdSchedules) {
+
+            nextDate = schedule.StartDate__c;
+
+            if (schedule.InstallmentPeriod__c == RD2_Constants.FREQUENCY_MONTHLY || schedule.InstallmentPeriod__c == RD2_Constants.FREQUENCY_FIRST_AND_FIFTEENTH) {
+                nextDate = Date.newInstance(
```

Comment:
`nextDate` calculation fails ("null pointer exception") when "Day of Month" is not provided. It is not required field and it can be null.